packages/opentelemetry/src/index.ts

```diff
@@ -220,8 +244,33 @@ export async function init(config: OpenTelemetryConfig) {
 export async function shutdown(): Promise<void> {
   if (tracerProvider) {
     await tracerProvider.shutdown();
+    tracerProvider = null;
   }
 }
 
+export async function instrumented<T>(
```

Comment:
This is the interesting bit, the other changes to this file were mostly to make it possible to test functionality here. I'm honestly surprised that this functionality isn't part of the core OTel API!

Reply:
I really like this kind of wrapper. I've been enjoying `runInTransaction()` recently which is similar.

Reply:
Yes! Me too. I with there was language-level support for this like Python's `with ...` syntax, but the function form isn't so bad either.

Reply:
That would be so nice!